src/common/helpers/escape-text.js

```diff
@@ -0,0 +1,35 @@
+export const escapeHtml = (unsafe) => {
+  return unsafe
+    ?.replaceAll('&', '&amp;')
+    .replaceAll('<', '&lt;')
+    .replaceAll('>', '&gt;')
+    .replaceAll('"', '&quot;')
+    .replaceAll("'", '&#39;')
+}
+
+export const escapeMarkdown = (unsafe) => {
+  return unsafe
+    ?.replaceAll('\\', '\\\\')
+    .replaceAll('`', '\\`')
+    .replaceAll('*', '\\*')
+    .replaceAll('_', '\\_')
+    .replaceAll('{', '\\{')
+    .replaceAll('}', '\\}')
+    .replaceAll('[', '\\[')
+    .replaceAll(']', '\\]')
+    .replaceAll('(', '\\(')
+    .replaceAll(')', '\\)')
+}
+
+export const escapeJsonValues = (data) => {
```

Comment:
We shouldn't be mutating the incoming data here (or almost anywhere, IMO) - it's a recipe for spooky-action-at-a-distance surprises.

Something along the lines of the following would work without that:

```javascript
export const escapeJsonValues = (data) => {
  let escapedData = {}
  Object.keys(data).forEach((key) => {
    if (typeof data[key] === 'string') {
      escapedData[key] = escapeHtml(data[key])
    } else if (typeof data[key] === 'object' && data[key] !== null) {
      escapedData[key] = escapeJsonValues(data[key])
    } else {
       escapedData[key] = data[key]
    }
  })
  return escapedData
}
```